src/main/java/org/example/persistence/PersistenceProgram.java

```diff
@@ -1,54 +1,422 @@
+
+        try {
+            PreparedStatement statement = connection.prepareStatement("DELETE program, programinformation FROM program INNER JOIN " +
+                    "programinformation where program.id = ? and programinformation.id = program.programinformation");
+            statement.setLong(1,programEntity.getId());
+            return statement.execute();
+        } catch (SQLException e) {
+            e.printStackTrace();
+            return false;
+        }
+
+    }*/
+
+
+    public boolean softDeleteProgramTable(ProgramEntity programEntity){
+        try {
+            PreparedStatement statement =connection.prepareStatement("UPDATE program SET timestamp_for_deletion = ? " +
+                    "WHERE id = ? ");
+
+            statement.setTimestamp(1, new Timestamp(System.currentTimeMillis()));
+            statement.setLong(2,programEntity.getId());
+            statement.executeUpdate();
+            //connection.commit();
+            return true;
+        } catch (SQLException e) {
+            e.printStackTrace();
+            return false;
+        }
+
+    }
+
 
     @Override
     public boolean deleteProgram(ProgramEntity programEntity) {
-        return false;
+        softDeleteProgramTable(programEntity);
+
+
+        return true;
+
     }
 
     @Override
     public boolean updateProgram(ProgramEntity programEntity) {
-        return false;
+        try {
+            PreparedStatement stmt = connection.prepareStatement("UPDATE  programinformation SET title = ?, description = ? " +
+                    " WHERE programinformation.id = ? ");
+            stmt.setString(1,programEntity.getName());
+            stmt.setString(2,programEntity.getDescription());
+            stmt.setLong(3,programEntity.getPrograminformationId());
+            stmt.executeUpdate();
+
+            return true;
+
+        } catch (SQLException e) {
+            e.printStackTrace();
+            return false;
+        }
+
+
+    }
+
+    public ArrayList<CreditEntity> getCreditUser(ProgramEntity programEntity){
+        try {
+            PreparedStatement statement = connection.prepareStatement("SELECT credit.program, \"user\".id, \"user\".title, \"user\"" +
+                    ".firstname, \"user\".middlename ,\"user\".lastname,\"user\".email from credit inner join " +
+                    "\"user\" on credit.\"user\" = \"user\".id where credit.program = ? ");
+            statement.setLong(1,programEntity.getId());
+            ResultSet resultSet = statement.executeQuery();
+            ArrayList<CreditEntity> creditList = new ArrayList<>();
+            while (resultSet.next()){
+               CreditEntity credit = new CreditEntity(programEntity.getId(),createUserEntityFromResultSet(resultSet));
+               creditList.add(credit);
+            }
+            return creditList;
+        } catch (SQLException e) {
+            e.printStackTrace();
+            return null;
```

Comment:
Hvorfor beder du om et helt object, når du kun bruger id'et for det? hvad er tanken bag det?